mesonbuild/modules/cmake.py

```diff
@@ -256,31 +266,25 @@ def detect_cmake(self, state):
         self.cmake_detected = True
         return True
 
-    @permittedKwargs({'version', 'name', 'compatibility', 'install_dir'})
-    def write_basic_package_version_file(self, state, _args, kwargs):
-        version = kwargs.get('version', None)
-        if not isinstance(version, str):
-            raise mesonlib.MesonException('Version must be specified.')
-
-        name = kwargs.get('name', None)
-        if not isinstance(name, str):
-            raise mesonlib.MesonException('Name not specified.')
-
-        compatibility = kwargs.get('compatibility', 'AnyNewerVersion')
-        if not isinstance(compatibility, str):
-            raise mesonlib.MesonException('compatibility is not string.')
-        if compatibility not in COMPATIBILITIES:
-            raise mesonlib.MesonException('compatibility must be either AnyNewerVersion, SameMajorVersion or ExactVersion.')
+    @typed_kwargs(
+        'cmake.write_basic_package_version_file',
+        KwargInfo('compatibility', str, default='AnyNewerVersion', validator=in_set_validator(set(COMPATIBILITIES))),
+        KwargInfo('install_dir', (str, NoneType), default=None),
+        KwargInfo('name', str, required=True),
+        KwargInfo('version', str, required=True),
+    )
+    def write_basic_package_version_file(self, state, _args, kwargs: 'WriteBasicPackageVersionFile'):
```

Comment:
`args` got renamed to a leading underscore, this is not the usual Meson style even for unused arguments.

On the other hand, this does conveniently point out that there are no allowed args, so we can raise an error message if people try to pass args anyway. See the `@noPosargs` decorator (from the same place `@typed_kwargs` is imported from).

Reply:
I didn't even realize, my IDE probably did it without even asking. But thanks I'll add the `@noPosargs` decorator

Reply:
Oh my gosh, lol. I wonder why... but it isn't your fault then. :D 

`@noPosargs` is still useful either way.